### ISSUE TITLE
Swift project Build Error when archiving

### ISSUE THREAD

**Issue by sunnyzhaog** [grpc/grpc#7406]
my project build  is no problem 

but when i try to  Release  app use Xcode archive .  i got this error  .  

<img width="650" alt="2016-07-15 8 25 25" src="https://cloud.githubusercontent.com/assets/20331892/16874157/5268a304-4aca-11e6-9c9c-59b80d3b6a33.png">

``` ruby
Pod::Spec.new do |s|
  s.name             = "YiJianGRPC"
  s.version          = "0.1.6"
  s.summary          = "socket connection with grpc"
  s.ios.deployment_target = '8.0'

  s.platform     = :ios, '8.0'
  s.requires_arc = true

  s.source_files = 'Pod/Classes/*.{h,m}'
  src = "Pod/Classes/"
  dir = "YiJian"

  s.dependency "!ProtoCompiler-gRPCPlugin", "~> 0.14"


  pods_root = '../Pods'


  protoc_dir = "#{pods_root}/!ProtoCompiler"
  protoc = "#{protoc_dir}/protoc"
  plugin = "#{pods_root}/!ProtoCompiler-gRPCPlugin/grpc_objective_c_plugin"



  s.prepare_command = <<-CMD
    mkdir -p #{dir}

    #{protoc} \
       --plugin=protoc-gen-grpc=#{plugin} \
       --objc_out=#{dir} \
       --grpc_out=#{dir} \
       -I #{src} \
       -I #{protoc_dir} \
       #{src}/yijian.proto
  CMD

  s.subspec "Messages" do |ms|
    ms.source_files = "#{dir}/*.pbobjc.{h,m}", "#{dir}/**/*.pbobjc.{h,m}"
    ms.header_mappings_dir = dir
    ms.requires_arc = false
    ms.dependency "Protobuf"
    ms.pod_target_xcconfig = {
     'GCC_PREPROCESSOR_DEFINITIONS' => '$(inherited) GPB_USE_PROTOBUF_FRAMEWORK_IMPORTS=1',
   }
  end

  s.subspec "Services" do |ss|
    ss.source_files = "#{dir}/*.pbrpc.{h,m}", "#{dir}/**/*.pbrpc.{h,m}"
    ss.header_mappings_dir = dir
    ss.requires_arc = true
    ss.dependency "gRPC-ProtoRPC"
    ss.dependency "#{s.name}/Messages"
  end


end
```

**Comment by jcanizales** [comment]
I can reproduce this problem on our sample app at `src/objective-c/examples/Sample`, when using frameworks.

FBSDK apparently had the same problem half a year ago: http://stackoverflow.com/questions/33762511/framework-error-include-of-non-modular-header-inside-framework-module It's worth checking whether they did something that fixed it.

**Comment by sipdar** [comment]
http://stackoverflow.com/questions/24103169/swift-compiler-error-non-modular-header-inside-framework-module?rq=1
set  CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = YES


this  works for me

**Comment by jcanizales** [comment]
I've just confirmed that everything works when enabling that in the libraries that depend on RxLibrary: `GRPCClient` (the raw gRPC client), `ProtoRPC` (the client used by the protobuf generated code), and the generated code itself. It's less than ideal that it can't just be solved in the podspec for RxLibrary (I'll open an issue in Cocoapods for it), but it lets us move forward by modifying the above-mentioned podspecs.

**Comment by hsaliak** [comment]
@muxi @makdharma is this issue still relevant?

**Comment by jcanizales** [comment]
These would be the remaining steps:

* Check whether the workaround in #7448 can be reverted with the latest Cocoapods version.
* If not, open an issue on the Cocoapods repo to fix it, and one on gRPC (or reuse this one) to do the cleanup once it's fixed.

**Comment by jcanizales** [comment]
@hsaliak should we have issues for pending cleanup work?

**Comment by k7** [comment]
@jcanizales 

I have found another problem, CLANG_ALLOW_NON_MODULAR_INCLUDES_IN_FRAMEWORK_MODULES = YES will not works.

These steps will produce the problem.

1.  build a Cocoapods module in Objective-C, let's call it **BaseRPC**
2. build a Cocoapods module in Swift, which is dependent on **BaseRPC**, let's call it **SwiftRPC**
3. let's create a new Xcode project which using **SwiftRPC**.  

This project will build error when archiving. However, I have no idea how to fix it.

Thank you for reading.

**Comment by k7** [comment]
I solved it by using Carthage instead of Cocoapods.

**Comment by jcanizales** [comment]
@muxi can you reproduce that?

**Comment by muxi** [comment]
I am going to take a look later today

**Comment by muxi** [comment]
Tried using `src/objective-c/examples/Sample`, with a self-written Swift podspec as `SwiftRPC` and ProtoRPC as `BaseRPC`. Unable to reproduce this issue with this setting though; archive is normal.

**Comment by jcanizales** [comment]
Maybe `ProtoRPC` has some config that needs to be manually set in a user-made `BaseRPC`, though.

**Comment by muxi** [comment]
Just tried a user-made one and result is the same :)

**Comment by muxi** [comment]
@k7 We probably need some more information on your problem. Could you provide a link to the project with problem or any code available?

**Comment by s4x** [comment]
@muxi have same problem. Build is ok, but archive is not.

Errors:
<img width="717" alt="2018-01-24 13 56 54" src="https://user-images.githubusercontent.com/16644890/35328885-e49552c6-010e-11e8-9690-69f1e9ca25e9.png">

Starts with this line, and disappear when it is removed (magic):
<img width="725" alt="2018-01-24 13 57 04" src="https://user-images.githubusercontent.com/16644890/35328904-f9968c4e-010e-11e8-8de9-c2596fa3a3e0.png">

**Comment by xyyDDD** [comment]
I used Cocoapods to  add Grpc in my Object-C project,when I run the project ,the project is normal,grpc can work also. But when I archive the project, it finds errors like blew.
<img width="359" alt="e53b2c19-581c-43b9-b845-5acce3eb6e27" src="https://user-images.githubusercontent.com/32646843/35340129-4c60228e-015d-11e8-8b65-e1bba47291ff.png">

**Comment by muxi** [comment]
I'll take a look

**Comment by muxi** [comment]
I am still not able to reproduce this issue. Used `src/objective-c/examples/SwiftSample` (after some Swift-3 conversion in #12280). If you can isolate a project for this issue that would be very helpful.

**Comment by paulb777** [comment]
This issue reproduces with the Firestore-Example scheme from the workspace generated by https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/Example/Podfile. 

You can do an archive build or just a plain build for `Generic iOS Device`. 

More discussion and workarounds at https://github.com/firebase/firebase-ios-sdk/issues/898#issuecomment-374440332, GoogleCloudPlatform/ios-docs-samples#30 and https://github.com/GoogleCloudPlatform/ios-docs-samples/blob/master/speech/Swift/Speech-gRPC-Streaming/BUILDFIXES.

**Comment by muxi** [comment]
Thanks Paul. I'll take a look.

**Comment by muxi** [comment]
Update: The issue is confirmed. The actual solution seems to be more than trivial. Still working on it.

**Comment by stevenyix** [comment]
i was able to successfully create an archive after commenting out "#import "transformations/GRXMappingWriter.h" in gRPC-RxLibrary-umbrella.h

**Comment by muxi** [comment]
Fix (#15531) merged and will be in next release.
